### PR TITLE
Add some slash commands FAQs

### PR DESCRIPTION
**Pull request by PythonCoderAS**
## Summary

It adds some documentation that starts to list some common questions and answers for slash commands.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by ChesterWOV-DPY — CHANGES_REQUESTED** [review]

File: docs/faq.rst [hunk 1]
```diff
@@ -406,3 +406,16 @@ Example: ::
...
+
+Questions regarding Discord's new Application Commands.
+
+My bot's commands are not showing up!
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Did you :meth:`~.CommandTree.sync` your command? Commands need to be synced before they will appear.
+2. Did you invite your bot with the correct permissions? Bots need to be invited with the ``application.commands``
+   scope in addition to the ``bot`` scope. For example, invite the bot with the following URL:
+   https://discord.com/oauth2/authorize?client_id=<client id>&scope=applications.commands+bot
```
Comment (Rapptz) [inline docs/faq.rst:421]:
```suggestion
   ``https://discord.com/oauth2/authorize?client_id=<client id>&scope=applications.commands+bot``
```

This should also probably mention that `utils.oauth_url` helps create this URL too.

File: docs/faq.rst [hunk 2]
```diff
@@ -406,3 +406,16 @@ Example: ::
         await ctx.send(f'Pushing to {remote} {branch}')
 
 This could then be used as ``?git push origin master``.
+
+Application Commands
+--------------------
+
+Questions regarding Discord's new Application Commands.
```
Comment (Rapptz) [inline docs/faq.rst:413]:
```suggestion
Questions regarding Discord's new application commands, frequently known as "slash commands" or "context menu commands".
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
Just a few wording nits.

File: docs/faq.rst (hunk 2 above)
Comment (PythonCoderAS) [reply docs/faq.rst:413]: Slash commands and context menu commands are not alternate names for application commands, rather they are types of application commands.

Comment (Rapptz) [reply docs/faq.rst:413]: I don't see the relevance for the note. Most users will call them slash commands or context menu commands, that's what that note is saying. The API calls them application commands, yes, but users and developers just call them something else.

Comment (PythonCoderAS) [reply docs/faq.rst:413]: The problem is that **slash commands** and **context menu commands** _are not the same thing_.

Comment (Rapptz) [reply docs/faq.rst:413]: Sure. Except this has nothing to do with the note which is there to help users understand what an application command is from their perspective. This type of pedantry is unnecessary and is just holding back the merging of this PR.

Comment (PythonCoderAS) [reply docs/faq.rst:413]: That's a valid point, but I was worried about someone calling context menu commands "slash commands" or vice versa.

**Comment by PythonCoderAS** [comment]
Additionally, it might be time to change the default scopes for `oauth_url` to include the application commands scope as well.

**Comment by Rapptz** [comment]
That change was made a few weeks ago by now.

**Comment by PythonCoderAS** [comment]
> That change was made a few weeks ago by now.

Sorry, I was looking at a stale version of the docs, my bad.

**Review by Rapptz — APPROVED** [review]

**Pull request merged**